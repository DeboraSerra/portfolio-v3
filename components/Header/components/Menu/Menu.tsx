import { NextPage } from "next";
import Link from "next/link";
import { useRef, useState } from "react";
import { BiChevronDown, BiChevronUp } from "react-icons/bi";
import { useOnClickOutside } from "usehooks-ts";
import { ProjectsPaths } from "../../../../helpers/interfaces";
import ProjectsMenu from "../ProjectsMenu/ProjectsMenu";

interface Props {
  paths: ProjectsPaths[];
}

const Menu: NextPage<Props> = ({ paths }) => {
  const [openProject, setOpenProject] = useState(false);
  const projects = useRef<HTMLDivElement>(null);

  useOnClickOutside(projects, () => setOpenProject(false));

  const handleOpenProject = () => setOpenProject(!openProject);

  return (
    <div className="header__links">
      <Link href="/about" className="header__link">
        About Me
      </Link>
      <div onClick={handleOpenProject} className="header__link" ref={projects}>
        Projects {openProject ? <BiChevronUp /> : <BiChevronDown />}
        {openProject && <ProjectsMenu paths={paths} />}
      </div>
      <Link href="/contact" className="header__link">
        Contact
      </Link>
    </div>
  );
};

export default Menu;
