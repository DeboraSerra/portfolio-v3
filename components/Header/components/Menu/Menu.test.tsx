import { render } from "@testing-library/react";
import Menu from "./Menu";

describe("Test the Menu component", () => {
  it('Should match the snapshot', () => {
    const { asFragment } = render(<Menu />)
    expect(asFragment()).toMatchSnapshot();
  })
})