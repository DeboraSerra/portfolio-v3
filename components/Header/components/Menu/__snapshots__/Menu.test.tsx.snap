// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Menu component Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="header__links"
  >
    <a
      class="header__link"
      href="/about"
    >
      About Me
    </a>
    <div
      class="header__link"
    >
      Projects 
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
        />
      </svg>
    </div>
    <a
      class="header__link"
      href="/contact"
    >
      Contact
    </a>
  </div>
</DocumentFragment>
`;
